data/comments.js

```diff
@@ -0,0 +1,156 @@
+const mongoCollections = require('../config/mongoCollections')
+const recipes = require('./recipes');
+let { ObjectId } = require('mongodb');
+const recipesCo =  mongoCollections.recipes;
+const checkFunction = require('./verify');
+
+//not finished still need change in user collection
+async function createComment(recipeId,userId,text){
+    if(arguments.length!=3) throw "error number of arguments";
+    let newId = new ObjectId();
+    checkFunction.isCheckId(recipeId);
+    checkFunction.isCheckId(userId);
+    checkFunction.isCheckText(text);
+    const recipeThatComment = await recipes.getRecipeById(recipeId);
+    
+    const newComment = {
+        _id: newId,
+        recipeId:recipeId,
+        userId:userId,
+        text:text
+    };
+    //in this part I push comment into recipes collection 
+    //and we still need to do this in user collection
+    recipeThatComment.comments.push(newComment);
+    
+    //change String Id in comments into Object Id
+    let arr = recipeThatComment.comments;
+    
+
+    for(let i = 0; i < arr.length; i++){
+        for(let j in arr[i]){
+            if(j == '_id'){
+                arr[i][j] = ObjectId(arr[i][j]);
```

Comment:
Why do we need to convert string into Object ID?

Reply:
oh， i just copy it from lab6 , never mind